main/php_variables.c

```diff
@@ -404,16 +405,39 @@ static inline int add_post_vars(zval *arr, post_var_data_t *vars, bool eof)
 SAPI_API SAPI_POST_HANDLER_FUNC(php_std_post_handler)
 {
 	zval *arr = (zval *) arg;
-	php_stream *s = SG(request_info).request_body;
+	bool custom_input_stream = SG(request_parse_body_context).input_stream;
+	php_stream *s = custom_input_stream
+		? SG(request_parse_body_context).input_stream
+		: SG(request_info).request_body;
 	post_var_data_t post_data;
 
 	if (s && SUCCESS == php_stream_rewind(s)) {
+		ssize_t total_len = 0;
+		zend_long post_max_size = REQUEST_PARSE_BODY_OPTION_GET(post_max_size, SG(post_max_size));
+		bool throw_exceptions = SG(request_parse_body_context).throw_exceptions;
+
+#define EMIT_WARNING_OR_ERROR(...) do { \
+		if (throw_exceptions) { \
+			zend_throw_exception_ex(NULL, 0, __VA_ARGS__); \
+		} else { \
+			sapi_module.sapi_error(E_WARNING, __VA_ARGS__); \
+		} \
+	} while (0)
```

Comment:
Is it just me or do you also find this annoying for debugging. Personally I find macros a bit annoying if they don't wrap a single function call as it hides steps during debugging for me but might be just my lame debugging skills... :)

Reply:
Macros are indeed annoying for debugging. The macro avoids spelling out the local throw_exceptions variable repeatedly. We could achieve the same by adding a function and a macro that just passes the var. I can do that if you prefer.